mne/preprocessing/flat.py

```diff
@@ -9,10 +9,11 @@
 from ..io import BaseRaw
 from ..io.pick import _picks_to_idx
 from ..utils import (_validate_type, verbose, logger, _pl,
-                     _mask_to_onsets_offsets, ProgressBar)
+                     _mask_to_onsets_offsets, ProgressBar, deprecated)
 
 
 @verbose
+@deprecated('use mne.preprocessing.annotate_amplitude instead.')
 def annotate_flat(raw, bad_percent=5., min_duration=0.005, picks=None,
                   verbose=None):
     """Annotate flat segments of raw data (or add to a bad channel list).
```

Comment:
Ideally this function should now be a one-line wrapper that just calls `annotate_amplitude` now. That also helps test that no functionality has changed, because the existing `annotate_flat` tests should continue to work with no modification (other than dealing with the warning).

Reply:
Now that I added it, I remember why I didn't do it earlier. 
I changed some of the behavior of the function and fixed some bugs:

- **before**: `mean > bad_percent` -> mark entire channel as bad --- **after**: `mean >= bad_percent` -> mark entire channel as bad
Now, if you look for flat with `bad_percent=100`, and a channel is entirely flat, it will be set as bad instead of creating an annotation covering the entire raw. To me, this behavior made more sense.

- Fix a bug where annotations were off by 1 sample (a classic).
- Fix bugs with edge cases and overlap with `bad_acq_skip` (doesn't impact tests)

That's the 3 fixes I remember. For the 2 first, if I change the deprecated function with a one-line wrapper around `annotate_amplitude`, I also have to change the tests.. + now the deprecated function also has different behavior in some cases. You can find the changes in commit https://github.com/mne-tools/mne-python/pull/10143/commits/dbf5476577b268b6ec2949a92aee4c96102c2c07 but I would prefer to revert it.

Reply:
> Now, if you look for flat with bad_percent=100, and a channel is entirely flat, it will be set as bad instead of creating an annotation covering the entire raw. To me, this behavior made more sense.

Similarly, though, it means there is no longer a way to say "never change info['bads'], only ever annotate". I'm not sure about this change of behavior. I think I'd rather keep the old one. Then we can keep the one-line wrapper plus bugfixes, which seems nice/helpful rather than having code to do the same thing two different ways (even temporarily). I can live with it either way, but I'm +0.5 on reverting this change in behavior and keeping the one-line wrapper. @drammock WDYT?

Reply:
> I don't feel like this is needed because the annotations are not channel specific... Since the annotations are not channel specific, you now have a dataset with 80% of BAD_ annotation.

Thinking about the future, this doesn't have to be the case anymore -- I think it is this way only because the `annotate_flat` function predated the notion of channel-specific annotations. In the future we probably want the option to make these channel specific (and maybe it would become the default?). Combining that with the long-discussed-but-not-yet-implemented idea of epoch-specific bad channel interpolation (and probably also epoch annotations?) then the use cases for large `bad_percent` values (and even "never add to bads, only annotate") becomes pretty reasonable actually.

In any case, I guess it doesn't matter too much, so I can live with the solution where we revert  dbf5476 and add an option to have values > 100 to mean "never add to the `bads` returned value, only annotate".

Reply:
Agree, but I would hold on `bad_perent=None` setting until channel-specific annotation and epoch-specific bad channel interpolation + annotation are well implemented.